### PR TITLE
#106: Use VSCL

### LINKED ISSUE #106: Use `virtual-schema-common-lua`
## Situation

Recently we started extracting the common part for all Lua-based Virtual Schemas into a separate library. We should now use that library to prove it still works and to remove code duplication.

## Tasks

- [x] exasol/virtual-schema-common-lua#1
- [ ] Remove code duplication from this repository
- [ ] Remove specification duplication from this repository
- [ ] Add VSLC as LuaRocks dependency
- [ ] Get all tests green

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #106.

### CODE REVIEW

File: src/main/lua/exasolrls/MetadataReader.lua [hunk 1]
```diff
@@ -5,223 +5,205 @@ local exaerror = require("exaerror")
...
+        if self:_is_included_table(table_id, include_tables_lookup) and not self:_is_rls_metadata_table(table_id)
+        then
+            local columns, tenant_protected, role_protected, group_protected = self:_translate_columns_metadata(schema_id, table_id)
+            table.insert(tables, {name = table_id, columns = columns})
+            local protection = (tenant_protected and "t" or "-") .. (role_protected and "r" or "-")
                     .. (group_protected and "g" or "-")
-                log.debug('Found table "' .. table_id .. '" (' .. #columns .. ' columns). Protection: ' .. protection)
-                table.insert(table_protection, table_id .. ":" .. protection)
-            end
+            log.debug('Found table "' .. table_id .. '" (' .. #columns .. ' columns). Protection: ' .. protection)
```
Comment (kaklakariada) [inline src/main/lua/exasolrls/MetadataReader.lua:177]: Use placeholders instead of string concatenation

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: tools/runtests.sh [hunk 2]
```diff
@@ -32,33 +32,8 @@ function create_target_directories {
...
-    if [[ "$failures" -eq 0 ]]
-    then
-        echo -e "\e[1m\e[32m$failures failures\e[0m."
-        return "$exit_ok"
-    else
-        echo -e "\e[1m\e[31m$failures failures\e[0m."
-        return "$exit_software"
-    fi
+    cd "$base_dir" || exit
+    busted -c
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply tools/runtests.sh:36]: Done.

File: src/main/lua/exasolrls/RlsAdapter.lua [hunk 3]
```diff
@@ -1,61 +1,103 @@
...
+function RlsAdapter:get_version()
+    return VERSION
+end
+
+--- Get the name of the Virtual Schema adapter.
+--
+-- @return Virtual Schema adapter name
+--
+function RlsAdapter:get_name()
+    return "Row-level Security adapter (LUA)"
```
[the remark this answers is not in the document]
Comment (redcatbear) [reply src/main/lua/exasolrls/RlsAdapter.lua:34]: Done.

**Review by kaklakariada — DISMISSED** [review]

File: pom.xml [hunk 4]
```diff
@@ -229,29 +229,31 @@
                             <goal>exec</goal>
                         </goals>
                         <configuration>
-                            <executable>lua</executable>
+                            <executable>amalg.lua</executable>
```
Comment (kaklakariada) [inline pom.xml:232]: 👍

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**